backend-test/RequirementTests/FilterReqTests.hs

```diff
@@ -0,0 +1,149 @@
+{-|
+Description: Filter Req Data module tests.
+
+Module that contains the tests for the filtering of Req within the DynamicGraphs module.
+
+-}
+
+module RequirementTests.FilterReqTests
+( test_filterReqs ) where
+
+import Database.Requirement (Req (..))
+import DynamicGraphs.GraphGenerator (filterReq)
+import DynamicGraphs.GraphOptions (GraphOptions (..))
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (assertEqual, testCase)
+
+-- Test GraphOptions
+testGraphOptionsPopDept :: GraphOptions
+testGraphOptionsPopDept =
+    GraphOptions { taken = ["CSC108H1"],
+                departments = ["CSC", "STA"],
+                excludedDepth = 0,
+                maxDepth = -1,
+                courseNumPrefix = [],
+                distribution = [],
+                location = [],
+                includeRaws = True,
+                includeGrades = True
+                }
+
+testGraphOptionsEmptyDept :: GraphOptions
```

Comment:
Similarly, this can be `graphOptionsWithTaken` (the default is no take or departments).